### DISCUSSION TITLE
MQTT username/password?

### DISCUSSION THREAD

**Discussion by fokcuk** [SBFspot/SBFspot#473] [General]
Is it possible to provide username/password to MQTT connection?

**Comment by sbf-** [comment]
Yes, define mqtt_username and mqtt_password in config file

**Reply by fokcuk** [reply]
Did not work...
Reading config '/etc/sbfspot/SBFspot.cfg'
Warning: Ignoring keyword 'MQTT_Username'
Warning: Ignoring keyword 'MQTT_Password'

**Reply by sbf-** [reply]
I'm a bit confused now... I thought #445 was fixed, but it isn't.
So you'll have to use the workaround as described in #380

Add username and password to the MQTT_PublisherArgs option in SBFspot.cfg

```
MQTT_PublisherArgs=-u username -P password -h {host} -t {topic} -m "{{message}}"
```

**Reply by fokcuk** [reply]
thank you. it worked!
